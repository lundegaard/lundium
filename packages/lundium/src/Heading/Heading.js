import React from 'react';
import PropTypes from 'prop-types';
import { cx } from 'ramda-extension';

import Text from '../Text';

const Heading = ({ as = 'h2', children, className, ...rest }) => (
	<Text {...rest} className={cx(className)} as={as}>
		{children}
	</Text>
);

Heading.propTypes = {
	/** Tag used for root component */
	as: PropTypes.any,
	/** Children to be rendered in the main container. */
	children: PropTypes.node,
	/** Class name for root. */
	className: PropTypes.string,
	...Text.propTypes,
};

export default Heading;
