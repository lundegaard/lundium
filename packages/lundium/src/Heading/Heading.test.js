import React from 'react';
import { shallow } from 'enzyme';

import Heading from './Heading';

describe('<Heading />', () => {
	const requiredProps = {};

	it('shallow', () => {
		const wrapper = shallow(<Heading {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
