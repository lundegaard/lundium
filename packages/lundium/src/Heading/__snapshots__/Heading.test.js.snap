// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> shallow 1`] = `
<forwardRef(Text)
  as="h2"
  className=""
/>
`;
